app/uk/gov/hmrc/gform/models/PageShader.scala

```diff
@@ -52,12 +52,21 @@ class PageShader(
     val section = dynamicSections(sectionNumber.value)
     for {
       snippets <- Future.sequence(section.fields.map(f => htmlFor(f, 0)))
-      javascript = fieldJavascript(dynamicSections.flatMap(_.atomicFields(repeatService)))
+      javascript = createJavascript(dynamicSections.flatMap(_.fields), dynamicSections.flatMap(_.atomicFields(repeatService)))
       hiddenTemplateFields = dynamicSections.filterNot(_ == section).flatMap(_.atomicFields(repeatService))
       hiddenSnippets = Fields.toFormField(fieldData, hiddenTemplateFields, repeatService).map(formField => uk.gov.hmrc.gform.views.html.hidden_field(formField))
     } yield PageForRender(formId, sectionNumber, section.title, section.description, hiddenSnippets, snippets, javascript, envelopeId)
   }
 
+  private def createJavascript(fieldList: List[FieldValue], atomicFields: List[FieldValue]) = {
+    val groups: List[(FieldId, Group)] = fieldList.flatMap(fv => fv.`type` match {
```

Comment:
Daniel, there is nice `collect` function in scala collections which allow you to do something like this:
```
fieldList.collect{case g: Group => g)
```
which will result in List[Group]